src/Our.Umbraco.StackedContent/Web/UI/App_Plugins/StackedContent/js/stackedcontent.controllers.js

```diff
@@ -3,40 +3,40 @@
     "$scope",
     "editorState",
     "notificationsService",
+    "localStorageService",
     "innerContentService",
     "Our.Umbraco.StackedContent.Resources.StackedContentResources",
 
-    function ($scope, editorState, notificationsService, innerContentService, scResources) {
+    function ($scope, editorState, notificationsService, localStorageService, innerContentService, scResources) {
 
         $scope.inited = false;
         $scope.markup = {};
         $scope.prompts = {};
         $scope.model.value = $scope.model.value || [];
 
+        $scope.contentTypeGuids = _.uniq($scope.model.config.contentTypes.map(function (itm) {
+            return itm.icContentTypeGuid;
+        }));
+
         $scope.canAdd = function () {
-            return (!$scope.model.config.maxItems || $scope.model.config.maxItems === 0 || $scope.model.value.length < $scope.model.config.maxItems) && $scope.model.config.singleItemMode !== "1";
+            return (!$scope.model.config.maxItems || $scope.model.config.maxItems === "0" || $scope.model.value.length < $scope.model.config.maxItems) && $scope.model.config.singleItemMode !== "1";
         };
 
         $scope.canDelete = function () {
             return $scope.model.config.singleItemMode !== "1";
         };
 
         $scope.canCopy = function () {
-            var test = "test";
-            try {
-                window.localStorage.setItem(test, test);
-                window.localStorage.removeItem(test);
-                return true;
-            } catch (e) {
-                return false;
-            }
-		}
+            // TODO: Move this to InnerContent Service
+            return localStorageService.isSupported;
+        };
 
         $scope.canPaste = function () {
-            var stackedContentItem = JSON.parse(window.localStorage.getItem("StackedContentCopy"));
-            if (stackedContentItem && validateModel(stackedContentItem)) return true;
+            if ($scope.canCopy() && $scope.canAdd()) {
```

Comment:
Maybe canAdd should include a ls.isSupported check so you only need to check canAdd? Seems odd to check canCopy

Reply:
My thinking was that you can't paste without copying it first. But yeah, we can replace it with `isSupported`.